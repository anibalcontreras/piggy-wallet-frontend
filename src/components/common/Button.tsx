import React from 'react';
import { Text, Pressable, StyleSheet } from 'react-native';
import { Buttons, Sizing } from '../../styles';

interface ButtonProps {
  variant?: 'standard' | 'fullWidth';
  onPress?: () => void;
  children: string | string[];
}

function Button({ variant = 'standard', onPress, children }: ButtonProps): JSX.Element {
  let buttonStyle;
  switch (variant) {
    case 'standard':
      buttonStyle = style.standardButton;
      break;
    case 'fullWidth':
      buttonStyle = style.fullWidthButton;
      break;
  }

  return (
    <Pressable style={Buttons.applyOpacity(buttonStyle)} onPress={onPress}>
      <Text style={style.buttonText}>{children}</Text>
    </Pressable>
  );
}

const style = StyleSheet.create({
  standardButton: {
    ...Buttons.buttonStyle.standard,
    marginBottom: Sizing.x20,
  },
  fullWidthButton: {
    ...Buttons.buttonStyle.fullWidth,
    marginBottom: Sizing.x20,
  },
  buttonText: {
    ...Buttons.buttonText.primary,
  },
});

export default Button;
