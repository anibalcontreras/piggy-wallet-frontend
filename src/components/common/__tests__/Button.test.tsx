import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button';
import { Text } from 'react-native';

describe('Button Component', () => {
  it('renders standard button correctly', () => {
    const tree = renderer.create(<Button variant="standard">Test</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders fullWidth button correctly', () => {
    const tree = renderer.create(<Button variant="fullWidth">Test</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('displays the correct text', () => {
    const buttonText = 'Test Title';
    const tree = renderer.create(<Button>{buttonText}</Button>);
    const instance = tree.root;
    const textComponent = instance.findByType(Text);
    expect(textComponent.props.children).toBe(buttonText);
  });
});
