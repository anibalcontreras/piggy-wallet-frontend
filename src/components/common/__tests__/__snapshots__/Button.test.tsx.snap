// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders fullWidth button correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#01C38D",
      "borderRadius": 25,
      "justifyContent": "center",
      "marginBottom": 18,
      "opacity": 1,
      "padding": 18,
      "width": 724,
    }
  }
>
  <Text
    style={
      {
        "backgroundColor": "transparent",
        "color": "#FFFFFF",
        "fontFamily": "System",
        "fontSize": 16,
        "fontWeight": "700",
      }
    }
  >
    Test
  </Text>
</View>
`;

exports[`Button Component renders standard button correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#01C38D",
      "borderRadius": 25,
      "justifyContent": "center",
      "marginBottom": 18,
      "opacity": 1,
      "padding": 18,
      "width": 250,
    }
  }
>
  <Text
    style={
      {
        "backgroundColor": "transparent",
        "color": "#FFFFFF",
        "fontFamily": "System",
        "fontSize": 16,
        "fontWeight": "700",
      }
    }
  >
    Test
  </Text>
</View>
`;
